oracle/oracle.go

```diff
@@ -639,14 +636,18 @@ func (o *oracleOps) waitVolumeAttachmentStatus(volumeAttachmentID *string, desir
 			return nil, true, err
 		}
 		if getVolAttachmentResp.GetLifecycleState() == desiredStatus {
-			return nil, false, nil
+			return getVolAttachmentResp.GetDevice(), false, nil
 		}
 
 		logrus.Debugf("volume [%s] is still in [%s] state", *getVolAttachmentResp.GetVolumeId(), getVolAttachmentResp.GetLifecycleState())
 		return nil, true, fmt.Errorf("volume [%s] is still in [%s] state", *getVolAttachmentResp.GetVolumeId(), getVolAttachmentResp.GetLifecycleState())
 	}
-	_, err := task.DoRetryWithTimeout(f, cloudops.ProviderOpsTimeout, cloudops.ProviderOpsRetryInterval)
-	return err
+	devicePathRaw, err := task.DoRetryWithTimeout(f, cloudops.ProviderOpsTimeout, cloudops.ProviderOpsRetryInterval)
+	devicePath, ok := devicePathRaw.(*string)
+	if !ok {
```

Comment:
Instead of failing right away you can do a o.DevicePath() once and then error out.

		devicePath, err := o.DevicePath(volumeID)
		if err != nil {
			logrus.Errorf("Error while getting device path. Error: %v", err)
		}

Reply:
Done